### DISCUSSION TITLE
Prisma stored procedure call does not returns keys

### DISCUSSION THREAD

**Discussion by krishnans** [prisma/prisma#12312] [Q&A] [answered]
I have a stored procedure in mysql which does a select query against some tables.  I am executing this stored procedure using prisma raw query`call producer_name(${input})`.  Response from prisma does not have key name instead it look like this

[{
     f0: 123,
     f1: "fistname",
    f2: "lastname"
}]

Instead of field name I am getting f0, f1 & f2 etc.   Is this the internal design of prisma, or am I doing anything wrong here.  

I am using prisma client 3.10.0

EDIT:
Unfortunately I couldn't put my production stored procedure, so I have created a sample stored procedure you can use this in any mysql instance 

```
CREATE  PROCEDURE `test_procedure`()
BEGIN

select 1 as id,
"first name" as first_name,
"last_name" as last_name;
 
END

```

I have executed this and found the same issue.  below is the response I got from the prisma client
```
[
    {
        "f0": 1,
        "f1": "first name",
        "f2": "last_name"
    }
]

```
and query is
```
prisma.$queryRaw`call db_schema_name.test_procedure();`

```

My backend is nestjs based application

Thanks

**Comment by janpio** [comment]
What is the store procedure? How can we reproduce this behavior?

**Reply by krishnans** [reply]
Thanks,  I have updated my question with additional details, hope it helps.

**Comment by keepfit123** [comment]
？？any answer?

**Comment by RodrigoferroBR** [comment]
same trouble here... =(

**Comment by kikoken** [comment]
I have the same problem with using the call store procedure, not receiving the names' fields

**Comment by liahaHotmail** [comment]
i have the same issue, is it a bug/feature of prisma?

**Reply by anwar0908** [reply]
i believe this has been fixed in the current version 4.4.0

**Reply by Greatchinex** [reply]
@anwar0908 I am on the latest version and its still the same thing, Any way you got around this issue? Cause i really dont want to be doing `returnedValue[0].f1`

**Reply by pantharshit00** [reply]
Hello @Greatchinex,

Can you please report that here: https://github.com/prisma/prisma/issues/new?assignees=&labels=kind/bug&template=bug_report.yml ?

**Comment by anwar0908** [comment]
any progress on this issue yet ?

**Reply by janpio** [reply]
This is not an issue, but a GH discussion. If you have the same problem, please search if an actual issue exists and add your use case or 👍 reaction - or create a new one if it does not exist yet. Thanks.

**Comment by pantharshit00** [comment, marked as answer]
Anyone landing on this thread, please open a [new discussion](https://github.com/prisma/prisma/discussions/new) or a new [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) if you feel you are encountering unintended behaviour. Without relevant information, we won't be able to assist you so please open a new ticket. 

Related issue to this topic with more clarification: https://github.com/prisma/prisma/issues/9868#issuecomment-1108542174

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
